src/algebra/continued_fractions/computation/approximations.lean

```diff
@@ -0,0 +1,278 @@
+    simp [fract_lt_one, fract_nonneg, int_fract_pair.of, if_of_eq_ifp_n.symm] }
+end
+
+/-- Shows that the fractional parts of the stream are nonnegative. -/
+lemma fr_nonneg {ifp_n : int_fract_pair K}
+  (nth_stream_eq : int_fract_pair.stream v n = some ifp_n) :
+  0 ≤ ifp_n.fr :=
+(nth_stream_fr_nonneg_lt_one nth_stream_eq).left
+
+/-- Shows that the fractional parts of the stream smaller than one. -/
+lemma fr_lt_one {ifp_n : int_fract_pair K}
+  (nth_stream_eq : int_fract_pair.stream v n = some ifp_n) :
+  ifp_n.fr < 1 :=
+(nth_stream_fr_nonneg_lt_one nth_stream_eq).right
+
+/-- Shows that the integer parts of the stream are at least one. -/
+lemma one_le_succ_nth_stream_b {ifp_succ_n : int_fract_pair K}
+(succ_nth_stream_eq : int_fract_pair.stream v (n + 1) = some ifp_succ_n) :
+  1 ≤ ifp_succ_n.b :=
+begin
+  obtain ⟨ifp_n, nth_stream_eq, stream_nth_fr_ne_zero, ⟨refl⟩⟩ :
+    ∃ ifp_n, int_fract_pair.stream v n = some ifp_n ∧ ifp_n.fr ≠ 0
+    ∧ int_fract_pair.of ifp_n.fr⁻¹ = ifp_succ_n, from
+      succ_nth_stream_eq_some_iff.elim_left succ_nth_stream_eq,
+  change 1 ≤ ⌊ifp_n.fr⁻¹⌋,
+  suffices : 1 ≤ ifp_n.fr⁻¹, by { rw_mod_cast [le_floor], assumption },
+  have : 0 ≤ ifp_n.fr ∧ ifp_n.fr < 1, from nth_stream_fr_nonneg_lt_one nth_stream_eq,
+  cases this with ifp_n_fract_nonneg ifp_n_fract_lt_one,
+  suffices : 1 * ifp_n.fr ≤ 1, by
+  { rw inv_eq_one_div,
+    have : 0 < ifp_n.fr, from lt_of_le_of_ne ifp_n_fract_nonneg stream_nth_fr_ne_zero.symm,
+    solve_by_elim [le_div_of_mul_le], },
+  simp [(le_of_lt ifp_n_fract_lt_one)]
+end
+
+/--
+Shows that the `n + 1`th integer part `bₙ₊₁` of the stream is smaller or equal than the inverse of
+the `n`th fractional part `frₙ` of the stream.
+This result is straight-forward as `bₙ₊₁` is defined as the floor of `1 / frₙ`
+-/
+lemma succ_nth_stream_b_le_nth_stream_fr_inv {ifp_n ifp_succ_n : int_fract_pair K}
+  (nth_stream_eq : int_fract_pair.stream v n = some ifp_n)
+  (succ_nth_stream_eq : int_fract_pair.stream v (n + 1) = some ifp_succ_n) :
+  (ifp_succ_n.b : K) ≤ ifp_n.fr⁻¹ :=
+begin
+  suffices : (⌊ifp_n.fr⁻¹⌋ : K) ≤ ifp_n.fr⁻¹, by
+  { cases ifp_n with _ ifp_n_fr,
+    have : ifp_n_fr ≠ 0, by
+      { intro h, simpa [h, int_fract_pair.stream, nth_stream_eq] using succ_nth_stream_eq },
+    have : int_fract_pair.of ifp_n_fr⁻¹ = ifp_succ_n, by
+    { rw with_one.coe_inj.symm,
+      simpa [this, int_fract_pair.stream, nth_stream_eq] using succ_nth_stream_eq },
+    rwa ←this },
+  exact (floor_le ifp_n.fr⁻¹)
+end
+
+end int_fract_pair
+
+/-
+Next we translate above results about the stream of `int_fract_pair`s to the computed continued
+fraction `gcf.of`.
+-/
+
+/-- Shows that the integer parts of the continued fraction are at least one. -/
+lemma of_one_le_nth_part_denom {b : K}
+  (nth_part_denom_eq : (gcf.of v).partial_denominators.nth n = some b) :
+  1 ≤ b :=
+begin
+  obtain ⟨gp_n,  nth_s_eq, ⟨refl⟩⟩ : ∃ gp_n, (gcf.of v).s.nth n = some gp_n ∧ gp_n.b = b, from
```

Comment:
`refl` doesn't mean anything in a rcases pattern, only `rfl`

Reply:
Fixed